src/main/kotlin/io/emeraldpay/dshackle/upstream/error/UpstreamErrorHandler.kt

```diff
@@ -0,0 +1,25 @@
+package io.emeraldpay.dshackle.upstream.error
+
+import io.emeraldpay.dshackle.upstream.ChainRequest
+import io.emeraldpay.dshackle.upstream.Upstream
+import java.util.concurrent.CompletableFuture
+
+interface ErrorHandler {
+    fun handle(upstream: Upstream, request: ChainRequest, errorMessage: String?)
+
+    fun canHandle(request: ChainRequest, errorMessage: String?): Boolean
+}
+
+object UpstreamErrorHandler {
+    private val errorHandlers = listOf(
+        EthereumStateLowerBoundErrorHandler,
+    )
+
+    fun handle(upstream: Upstream, request: ChainRequest, errorMessage: String?) {
+        CompletableFuture.runAsync {
```

Comment:
runAsync runs code of default executor, could be problematic.

Reply:
Yes, it does, but why? Seems the submitted tasks are quite light to be executed on the default one

Reply:
And yet you're right. I was thinking a bit about it, and for sure it's better to avoid using default executors just because they can be used everywhere that can lead to unhappy consequences